### PR TITLE
TotemRPUVPattern is enabled as an input, replacing TotemRPRecHit

### PR DESCRIPTION
**Pull request by Cristian-Baldenegro**

### CODE REVIEW

**Comment by jan-kaspar** [comment]
Hi @Cristian-Baldenegro , yes, making a pull request is also a good way of integrating your changes. I will review your changes and let you know ~tomorrow.

File: Alignment/CTPPS/interface/StraightTrackAlignment.h [hunk 1]
```diff
@@ -22,6 +22,8 @@
 #include "DataFormats/CTPPSReco/interface/CTPPSDiamondRecHit.h"
 #include "DataFormats/CTPPSReco/interface/CTPPSPixelRecHit.h"
 
+#include "DataFormats/CTPPSReco/interface/TotemRPUVPattern.h"
+#include "DataFormats/CTPPSReco/interface/TotemRPLocalTrack.h"
```
Comment (jan-kaspar) [inline Alignment/CTPPS/interface/StraightTrackAlignment.h]: Do you need this include file?

File: Alignment/CTPPS/interface/StraightTrackAlignment.h [hunk 2]
```diff
@@ -55,7 +57,7 @@ class StraightTrackAlignment
 
     virtual void Begin(const edm::EventSetup&);
 
-    virtual void ProcessEvent(const edm::DetSetVector<TotemRPRecHit> &hitsStrip, const edm::DetSetVector<CTPPSDiamondRecHit> &hitsDiamond,
+    virtual void ProcessEvent(const edm::DetSetVector<TotemRPUVPattern> &UVPattern, const edm::DetSetVector<CTPPSDiamondRecHit> &hitsDiamond,
```
Comment (jan-kaspar) [inline Alignment/CTPPS/interface/StraightTrackAlignment.h]: Please use variable names starting with lower case. In this case please use a name that indicates its relation to strips, e.g. `uvPatternsStrip`.

File: Alignment/CTPPS/plugins/CTPPSFastLocalSimulation.cc [hunk 3]
```diff
@@ -3,7 +3,7 @@
 * This is a part of TOTEM offline software.
 * Authors: 
 *  Jan Kašpar (jan.kaspar@gmail.com) 
-*    
+*  Cristian Baldenegro (crisx.baldenegro@gmail.com)
```
Comment (jan-kaspar) [inline Alignment/CTPPS/plugins/CTPPSFastLocalSimulation.cc]: Nothing should be changed to `CTPPSFastLocalSimulation` - could you please revert your changes? Instead, we will add a module to the processing sequence with will convert rechits to UVPatterns.

File: Alignment/CTPPS/plugins/CTPPSStraightTrackAligner.cc [hunk 4]
```diff
@@ -35,7 +37,7 @@ class CTPPSStraightTrackAligner : public edm::EDAnalyzer
   private:
     unsigned int verbosity;
 
-    edm::EDGetTokenT<edm::DetSetVector<TotemRPRecHit>> tokenStripHits;
+    edm::EDGetTokenT<edm::DetSetVector<TotemRPUVPattern>> tokenPatternCollection; //newline
```
Comment (jan-kaspar) [inline Alignment/CTPPS/plugins/CTPPSStraightTrackAligner.cc]: Please use a name which indicates relation to strips.

File: Alignment/CTPPS/plugins/CTPPSStraightTrackAligner.cc [hunk 5]
```diff
@@ -61,7 +63,7 @@ using namespace edm;
 CTPPSStraightTrackAligner::CTPPSStraightTrackAligner(const ParameterSet &ps) : 
   verbosity(ps.getUntrackedParameter<unsigned int>("verbosity", 0)),
 
-  tokenStripHits(consumes<DetSetVector<TotemRPRecHit>>(ps.getParameter<edm::InputTag>("tagStripHits"))),
+  tokenPatternCollection(consumes<DetSetVector<TotemRPUVPattern>>(ps.getParameter<edm::InputTag>("tagUVPattern"))),
```
Comment (jan-kaspar) [inline Alignment/CTPPS/plugins/CTPPSStraightTrackAligner.cc]: Please use a tag name which indicates relation to strips.

File: Alignment/CTPPS/plugins/CTPPSStraightTrackAligner.cc [hunk 6]
```diff
@@ -96,8 +98,8 @@ void CTPPSStraightTrackAligner::analyze(const edm::Event &event, const edm::Even
   }
 
   // get input
-  Handle<DetSetVector<TotemRPRecHit>> inputStripHits;
-  event.getByToken(tokenStripHits, inputStripHits);
+  Handle<DetSetVector<TotemRPUVPattern>> inputPatternCollection;
```
Comment (jan-kaspar) [inline Alignment/CTPPS/plugins/CTPPSStraightTrackAligner.cc]: Please use a name which indicates relation to strips.

File: Alignment/CTPPS/python/ctppsStraightTrackAligner_cfi.py [hunk 7]
```diff
@@ -6,7 +6,7 @@
 
     # ---------- input and event selection ----------
 
-    tagStripHits = cms.InputTag("ctppsFastLocalSimulation"),
+    tagUVPattern = cms.InputTag("ctppsFastLocalSimulation"),
```
Comment (jan-kaspar) [inline Alignment/CTPPS/python/ctppsStraightTrackAligner_cfi.py]: Please use a tag name which indicates relation to strips.

File: Alignment/CTPPS/src/StraightTrackAlignment.cc [hunk 8]
```diff
@@ -315,7 +315,7 @@ void StraightTrackAlignment::Begin(const EventSetup &es)
 
 //----------------------------------------------------------------------------------------------------
 
-void StraightTrackAlignment::ProcessEvent(const DetSetVector<TotemRPRecHit> &hitsStrip, const DetSetVector<CTPPSDiamondRecHit> &hitsDiamond,
+void StraightTrackAlignment::ProcessEvent(const DetSetVector<TotemRPUVPattern> &UVPattern, const DetSetVector<CTPPSDiamondRecHit> &hitsDiamond,
```
Comment (jan-kaspar) [inline Alignment/CTPPS/src/StraightTrackAlignment.cc]: Please use a name which indicates relation to strips.

File: Alignment/CTPPS/src/StraightTrackAlignment.cc [hunk 9]
```diff
@@ -327,20 +327,61 @@ void StraightTrackAlignment::ProcessEvent(const DetSetVector<TotemRPRecHit> &hit
...
+        case TotemRPUVPattern::projV:
+          n_V++;
+          idx_V=pi;
+          break;
+
+        default:
+          break;
+       }  
+    }
+    //TODO Uncomment for real data.
```
Comment (jan-kaspar) [inline Alignment/CTPPS/src/StraightTrackAlignment.cc]: Please remove this block. We will always use only events with 1U and 1V pattern.

File: Alignment/CTPPS/src/StraightTrackAlignment.cc [hunk 10]
```diff
@@ -327,20 +327,61 @@ void StraightTrackAlignment::ProcessEvent(const DetSetVector<TotemRPRecHit> &hit
...
+    //    if (n_U != 1 || n_V != 1)
+    //    {
+    //	  std::cout<< ">> StraightTrackAlignment::ProcessEvent > Impossible to combine U and V patterns in RP " << detId << " (n_U=" << n_U << ", n_V=" << n_V << ")." << std::endl;
+    // continue;
+    //    }
+    
+    if (!pv[idx_U].getFittable() || !pv[idx_V].getFittable())
+	 continue; //For FastSimulation remove continue statement
```
Comment (jan-kaspar) [inline Alignment/CTPPS/src/StraightTrackAlignment.cc]: Please remove this comment. Here, there should be no difference data/simulation.

File: Alignment/CTPPS/src/StraightTrackAlignment.cc [hunk 11]
```diff
@@ -327,20 +327,61 @@ void StraightTrackAlignment::ProcessEvent(const DetSetVector<TotemRPRecHit> &hit
...
+
+    // combine U and V hits
+    DetSetVector<TotemRPRecHit> hits;
+    
+    for (auto &ids : pv[idx_V].getHits())
```
Comment (jan-kaspar) [inline Alignment/CTPPS/src/StraightTrackAlignment.cc]: There should be a similar loop over the U pattern, but I can't see it.

File: Alignment/CTPPS/interface/StraightTrackAlignment.h (hunk 1 above)
Comment (Cristian-Baldenegro) [inline Alignment/CTPPS/interface/StraightTrackAlignment.h]: Do you mean TotemRPLocalTrack.h? I was going to use that one at the beginning, but forgot to remove the include.

File: Alignment/CTPPS/plugins/CTPPSFastLocalSimulation.cc (hunk 3 above)
Comment (Cristian-Baldenegro) [inline Alignment/CTPPS/plugins/CTPPSFastLocalSimulation.cc]: Yes, I will revert the changes. I modified CTPPSFastLocalSimulation.cc to test the UV Pattern input.

**Pull request merged**